core/pom.xml

```diff
@@ -107,6 +107,11 @@
         </dependency>
 
         <dependency>
+            <groupId>org.jooq</groupId>
+            <artifactId>jool</artifactId>
```

Comment:
Just played with it. Not sure it solves the same problem. From what I can gather the try stuff lets you wrap checked lambdas into a try class. That's not helpful when you want to do al ambda in a pipeline that only accepts a certain functional interface. A good example is 

```
IntStream.range(...).mapToObj(() -> { // checked })
```

That lambda is of type `IntConsumer` and stupidly doesn't subclass just generic `Consumer`. So you can't pass in other consumer types.  j00q provides unchecked versions of all the default java types for this kind of shit.

That said, I _LOOOOVE_ the idea of wrapping our own checked exceptions that when need to be executed as lambdas can be wrapped in try's and composed/mapped with eithers.

Maybe i'm misunderstanding how javaslang does unchecked though. Open to discuss. I'll do this in the meantime and add it as a dependency so we can start playing with it :)